src/cookbooks/data_modeling/aggregates_guidelines.md

```diff
@@ -0,0 +1,91 @@
+
+- [Store the business logic in BigQuery](https://docs.telemetry.mozilla.org/cookbooks/data_modeling/where_to_store), ideally in a client_id level table to aggregate from.
+- All aggregates are version controlled in the git repositories `bigquery-etl` and `spoke-default`.
+- All types of aggregates require a backfill when the source data changes.
+  - The managed-backfill process can be used for aggregate tables.
+  - The materialized views are backfilled by ?. Any schema change in the base table invalidates the view and requires backfill.
+  - Looker PDTs require following a [protocol to backfill](https://mozilla.udemy.com/course/looker-training-for-developers/learn/lecture/35440216/), currently not automated.
+- Using indices, partition and clustering is allowed for all cases. Looker requires that they are defined in the base table.
+- There is a limit of 20 materialized views per table.
+
+
+## What are the use cases to use each of these aggregates?
+
+#### Use a BigQuery aggregate table when:
+- The results should not be unexpectedly affected by shredded.
+- The metric requires strict change control.
+- A visible alerts is required in case of failure or data out of sync. In this option, the scheduler sends email and alerts attended daily by the Data Engineering team during the Airflow Triage process.
+- The table will be queried directly or used as a source for other analysis. Looker PDTs are not designed to be queried directly.
+- The query requires full flexibility to use DML, data types, aggregation functions and different types of JOIN.
+
+#### Use a Materialized View when:
+- The results can be based on shredded data (tables with client_id).
+- The view will be queried directly or used as a source for other analysis.
+- Change control is **not** required or is already implemented in the base table. This can be verified by looking for the label `change_controlled: true` in the table's metadata.
+- An explicit alert on failure is **not** required. Failures need to be intentionally monitored in the Dashboard ?
+- The metric does **not** require non-deterministic functions that are not supported: RAND(), CURRENT_TIMESTAMP, CURRENT_DATE(), or CURRENT_TIME().
+- The query does **not** require UDFs, UNNESTING arrays, COUNT DISTINCT, ORDER BY or any DML operation different from SELECT.
+- The query uses a WITH clause, COUNTIF, INNER JOIN or TIMESTAMP_ADD. These are all supported.
+
+#### Use a Looker PDT when:
+- The results can be based on shredded data (tables with client_id).
+- Change control is not required or is already implemented in the base table. This can be verified by looking for the label `change_controlled: true` in the metadata.
+- An explicit alert on failure is **not** required. Failures are monitored in the Dashboard [Errors and Broken Content Dashboard](https://mozilla.cloud.looker.com/dashboards/system__activity::errors_and_broken_content) or in the [PDTs Dashboard](https://mozilla.cloud.looker.com/admin/pdts).
```

Comment:
@scholtzan what do you think about adding this dashboard to the Airflow Triage, since there are many PDT failures that don't generate an alert.

Reply:
Yeah, that makes sense. We could add a link to the health dashboard

Reply:
Sounds good, I added the link to PDTs admin to the [Platform Health Check](https://mozilla.cloud.looker.com/dashboards/387) so these are checked once a week.